datafusion/core/src/datasource/listing/url.rs

```diff
@@ -46,37 +46,54 @@ pub struct ListingTableUrl {
 impl ListingTableUrl {
     /// Parse a provided string as a `ListingTableUrl`
     ///
+    /// A URL can either refer to a single object, or a collection of objects with a
```

Comment:
This isn't changing the pre-existing behaviour but attempting to better document it